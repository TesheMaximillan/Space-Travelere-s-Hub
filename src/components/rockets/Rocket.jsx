/* eslint-disable react/prop-types */
import React from 'react';
import { useDispatch } from 'react-redux';
import { updateRocketStatus } from '../../redux/rockets/activities';
import StatusRes from './StatusRes';
import styles from '../../routes/rockets.module.scss';

function Rocket({ rocket }) {
  const dispatch = useDispatch();
  const {
    rocketsImg, rocketName, rocketId, description, status,
  } = rocket;

  const handleStatus = () => dispatch(updateRocketStatus(rocketId, !status));
  return (
    <div key={rocketId} className={styles.rocketsItems}>
      <img className={styles.rocketsImg} src={rocketsImg} alt="Rocket setoff" />
      <div>
        <span className={styles.rocketsTitle}>
          {rocketName}
        </span>
        <StatusRes description={description} status={status} handleStatus={handleStatus} />
      </div>
    </div>
  );
}

export default Rocket;
