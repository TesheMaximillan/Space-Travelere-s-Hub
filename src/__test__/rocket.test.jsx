/* eslint-disable no-whitespace-before-property */
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rocket from '../components/rockets/Rocket';

const RockeT = {
  rocketImg: 'rocketsImg',
  rocketName: 'rocketName',
  rocketId: 'rocketId',
  description: 'description',
  status: 'true',
};

it('renders correctly', () => {
  const tree = renderer .create(
    <Provider store={store}>
      <Rocket key={1} rocket={RockeT} />
      {' '}

    </Provider>,
  ) .toJSON();
  expect(tree).toMatchSnapshot();
});
