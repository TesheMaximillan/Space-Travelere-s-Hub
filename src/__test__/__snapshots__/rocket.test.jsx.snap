// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="rocketsItems"
  >
    <img
      alt="Rocket setoff"
      className="rocketsImg"
    />
    <div>
      <span
        className="rocketsTitle"
      >
        rocketName
      </span>
      <span>
        <span
          className="statusActive"
        >
          Reserved
        </span>
        <span
          className="rocketsDescr"
        >
          description
        </span>
      </span>
      <div>
        <button
          className="btnRocket btnLeave"
          onClick={[Function]}
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </div>
  </div>,
  " ",
]
`;
